### DISCUSSION TITLE
LGPL Licensing question

### DISCUSSION THREAD

**Discussion by bodand** [rdbende/Azure-ttk-theme#8] [Question]
Hello there!

Firstly, good job on the theme; looks really good 😄 

The problem is that I don't know how to interpret the licensing requirements of the LGPL on a programing
language (Tcl) that doesn't understand the concept of linking - for other Tcl scripts at least.

The LGPL says this:

```
  5. A program that contains no derivative of any portion of the
Library, but is designed to work with the Library by being compiled or
linked with it, is called a "work that uses the Library".  Such a
work, in isolation, is not a derivative work of the Library, and
therefore falls outside the scope of this License.

  However, linking a "work that uses the Library" with the Library
creates an executable that is a derivative of the Library (because it
contains portions of the Library), rather than a "work that uses the
library".  The executable is therefore covered by this License.
Section 6 states terms for distribution of such executables.
```

However, a Tcl script (the Ttk theme) is only a runtime requirement of the Tk GUI to run and does not participate in the (non-existent) compiling or linking step. Furthermore, the Library is not compiled or linked in any shape or form and is only (re)distributed only in source form. Thinking about it this way, only section 1 of the LGPL applies at all times:

```
  1. You may copy and distribute verbatim copies of the Library's
complete source code as you receive it, in any medium, provided that
you conspicuously and appropriately publish on each copy an
appropriate copyright notice and disclaimer of warranty; keep intact
all the notices that refer to this License and to the absence of any
warranty; and distribute a copy of this License along with the
Library.
[...]
```

My question is whether my conclusion aligns with the copyright owner's interpretation of the license, and if not, how is it intended to be understood.

If a more concrete example of my use-case is needed, I have a BSD 3-Clause licensed project that uses Tcl/Tk for the GUI parts, in which I'd like to use this theme; the question is whether I can do that and just distribute the theme's files (license included, of course) with the project without having to relicense it as GPL. (In which case I'm just going to go scavenge for some other theme.)

Thanks in advance,
bodand

edit:
I just realized GitHub's new beta discussions are more appropriate for this, but oh well. It's too late now. Hope it's not much of a bother.

**Comment by rdbende** [comment]
First of all, thank you for trying out the theme.

Yes, you can use the theme in this use case, because that's how I interpret the license.

**Comment by mcuntz** [comment]
Dear rdbende,

this is really a good looking theme.

I would like to use it in a GUI that I am developing [ncvue](https://github.com/mcuntz/ncvue). I quite like the look of _ncvue_ on macOS but it is very ugly on Windows. So your scheme would be perfect. I would acknowledge it accordingly, of course.
I publish my software as open as possible and hence opted for the MIT License.

But your scheme is published under the license LGPL 2.1. It is most appropriate for libraries. It is very confusing to me, and to @boband as well I guess, how it applies to programs. A GUI that uses the azure scheme will most probably be a program.

I like the section [Differences from the GPL](https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License#Differences_from_the_GPL) on Wikipedia.
This is how I understood it:

1. If I publish the source code of the GUI, then it is o.k. to use a copy of the azure scheme, even if the GUI has another license.
2. If I produce a compiled version of the GUI, for example for Windows users, which in terms of a library would be called static linking, then it is still o.k. to use a copy as long as the source code is still publicly available.

Do you think that this is correct?

Kind regards,
Matthias

**Reply by rdbende** [reply]
Yes. I think it falls under "work that uses the Library."

**Reply by mcuntz** [reply]
Great. So I will use on Windows and test it also on the macOS and Linux.
Thanks
Matthias

**Comment by rdbende** [comment]
I relicensed the theme. As of version 2.0, the theme is available under the MIT license.
